lib/Dialect/SV/Ops.cpp

```diff
@@ -113,6 +114,54 @@ static void printModportStructs(OpAsmPrinter &p, Operation *,
   p << ')';
 }
 
+//===----------------------------------------------------------------------===//
+// Extract operation.
+//===----------------------------------------------------------------------===//
+
+static ParseResult parseExtractOp(OpAsmParser &p, OperationState &result) {
+  MLIRContext *ctxt = result.getContext();
+  FlatSymbolRefAttr field;
+  OpAsmParser::OperandType extractable;
+  Type inputType;
+  if (p.parseAttribute(field) || p.parseOperand(extractable) || p.parseColon())
+    return failure();
+  result.addAttribute("field", field);
+
+  auto typeLoc = p.getCurrentLocation();
+  if (p.parseType(inputType))
+    return failure();
+
+  if (p.resolveOperand(extractable, inputType, result.operands))
+    return failure();
+
+  // The type specified is the extractable type, so the return type can be
+  // inferred based on that. Said resolution is different depending on the
+  // extractable type.
+  if (auto ifaceTy = inputType.dyn_cast<InterfaceType>()) {
+    SymbolRefAttr ifaceSym = ifaceTy.getInterface();
+    SmallVector<FlatSymbolRefAttr, 4> refs;
+    auto nested = ifaceSym.getNestedReferences();
+    refs.append(nested.begin(), nested.end());
+    refs.push_back(field);
+
+    SymbolRefAttr modportSym =
+        SymbolRefAttr::get(ifaceSym.getRootReference(), refs, ctxt);
+    auto resultPort = InterfaceModportType::get(ctxt, modportSym);
+    result.addTypes({resultPort});
+    return success();
+  } else {
```

Comment:
super nit but no need for an 'else' after a return.  

I would also recommend making the happy path be the fall through with something like:

```
auto ifaceTy = inputType.dyn_cast<InterfaceType>();
if (!ifaceTy) {
  error stuff
}

happy path
```